shared/constants/network.ts

```diff
@@ -1,16 +1,18 @@
+import { CaipChainId } from '@metamask/utils';
 import { capitalize, pick } from 'lodash';
 /**
  * A type representing any valid value for 'type' for setProviderType and other
  * methods that add or manipulate networks in MetaMask state.
  */
 export type NetworkType = (typeof NETWORK_TYPES)[keyof typeof NETWORK_TYPES];
 
+// Still needed?
 /**
  * A union type of all possible hard-coded chain ids. This type is not
- * exhaustive and cannot be used for typing chainId in areas where the user or
- * dapp may specify any chainId.
+ * exhaustive and cannot be used for typing caipChainId in areas where the user or
+ * dapp may specify any caipChainId.
  */
-export type ChainId = (typeof CHAIN_IDS)[keyof typeof CHAIN_IDS];
+// export type ChainId = (typeof CHAIN_IDS)[keyof typeof CHAIN_IDS];
```

Comment:
CaipChainId should be replacing this type. Unless we specifically meant this explicit subset of chain id